### PR TITLE
Deprecate 'remove_binary' default of false for ingest attachment processor

### PR DESCRIPTION
**Pull request by dakrone**
This commit adds deprecation warning for when the `remove_binary` setting is unset. In the future we want to change the default to `true` (it is currently `false`), so this will let a user know they should be explicit about setting this to ensure the behavior does not change in a future (breaking) release.

Relates to #86014

### CODE REVIEW

**Comment by dakrone** [comment]
@elasticmachine update branch

File: modules/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java [hunk 1]
```diff
@@ -235,7 +249,17 @@ public AttachmentProcessor create(
...
             String indexedCharsField = readOptionalStringProperty(TYPE, processorTag, config, "indexed_chars_field");
-            boolean removeBinary = readBooleanProperty(TYPE, processorTag, config, "remove_binary", false);
+            Boolean removeBinary = readOptionalBooleanProperty(TYPE, processorTag, config, "remove_binary");
+            if (removeBinary == null) {
+                DEPRECATION_LOGGER.warn(
+                    DeprecationCategory.PARSING,
+                    "attachment-remove-binary",
+                    "The default [remove_binary] value of 'false' is deprecated and will be "
+                        + "set to 'true' in a future release. Set [remove_binary] explicitly to "
+                        + "'true' or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java:259]:
```suggestion
                        + "'true' or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java [hunk 2]
```diff
@@ -54,6 +60,12 @@ public void testConfigureIndexedChars() throws Exception {
         assertThat(processor.getTag(), equalTo(processorTag));
         assertThat(processor.getIndexedChars(), is(indexedChars));
         assertFalse(processor.isIgnoreMissing());
+
+        assertWarnings(
+            "The default [remove_binary] value of 'false' is deprecated "
+                + "and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true'"
+                + " or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java:67]:
```suggestion
                + " or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java [hunk 3]
```diff
@@ -64,6 +76,12 @@ public void testBuildTargetField() throws Exception {
         assertThat(processor.getField(), equalTo("_field"));
         assertThat(processor.getTargetField(), equalTo("_field"));
         assertFalse(processor.isIgnoreMissing());
+
+        assertWarnings(
+            "The default [remove_binary] value of 'false' is deprecated "
+                + "and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true'"
+                + " or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java:83]:
```suggestion
                + " or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java [hunk 4]
```diff
@@ -82,6 +100,12 @@ public void testBuildFields() throws Exception {
         assertThat(processor.getField(), equalTo("_field"));
         assertThat(processor.getProperties(), equalTo(properties));
         assertFalse(processor.isIgnoreMissing());
+
+        assertWarnings(
+            "The default [remove_binary] value of 'false' is deprecated "
+                + "and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true'"
+                + " or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java:107]:
```suggestion
                + " or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java [hunk 5]
```diff
@@ -108,6 +132,12 @@ public void testBuildIllegalFieldOption() throws Exception {
         } catch (ElasticsearchParseException e) {
             assertThat(e.getMessage(), equalTo("[properties] property isn't a list, but of type [java.lang.String]"));
         }
+
+        assertWarnings(
+            "The default [remove_binary] value of 'false' is deprecated "
+                + "and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true'"
+                + " or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java:139]:
```suggestion
                + " or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java [hunk 6]
```diff
@@ -123,6 +153,12 @@ public void testIgnoreMissing() throws Exception {
         assertThat(processor.getTargetField(), equalTo("attachment"));
         assertThat(processor.getProperties(), sameInstance(AttachmentProcessor.Factory.DEFAULT_PROPERTIES));
         assertTrue(processor.isIgnoreMissing());
+
+        assertWarnings(
+            "The default [remove_binary] value of 'false' is deprecated "
+                + "and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true'"
+                + " or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorFactoryTests.java:160]:
```suggestion
                + " or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/20_attachment_processor.yml [hunk 7]
```diff
@@ -41,6 +47,8 @@
           wait_for_status: green
 
   - do:
+      warnings:
+        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/20_attachment_processor.yml:51]:
```suggestion
        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/20_attachment_processor.yml [hunk 8]
```diff
@@ -75,6 +83,8 @@
 ---
 "Test indexed chars are configurable":
   - do:
+      warnings:
+        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/20_attachment_processor.yml:87]:
```suggestion
        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/20_attachment_processor.yml [hunk 9]
```diff
@@ -110,6 +120,8 @@
 ---
 "Test indexed chars are configurable per document":
   - do:
+      warnings:
+        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/20_attachment_processor.yml:124]:
```suggestion
        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure no behavior change."
```

File: modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/30_files_supported.yml [hunk 10]
```diff
@@ -1,9 +1,15 @@
...
+  - skip:
+      features: warnings
+---
 "Test ingest attachment processor with .doc file":
   - skip:
       version: " - 7.99.99"
       reason: "new fields added in 8.0.0"
   - do:
+      warnings:
+        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure to behavior change."
```
Comment (masseyke) [inline modules/ingest-attachment/src/yamlRestTest/resources/rest-api-spec/test/ingest_attachment/30_files_supported.yml:12]:
```suggestion
        - "The default [remove_binary] value of 'false' is deprecated and will be set to 'true' in a future release. Set [remove_binary] explicitly to 'true' or 'false' to ensure no behavior change."
```

**Review by masseyke — APPROVED** [review]
LGTM (assuming that test failure is unrelated).

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Pull request merged**